crates/proof-of-sql/src/sql/parse/query_expr.rs

```diff
@@ -62,10 +65,20 @@ impl<C: Commitment> QueryExpr<C> {
         let group_by = context.get_group_by_exprs();
         if !group_by.is_empty() {
             if let Some(group_by_expr) = Option::<GroupByExpr<C>>::try_from(&context)? {
+                // If the group by expression is provable the projection step is just identity.
```

Comment:
I do worry about this a little bit - we're assuming this is the case because of the `Some` result of the `GroupByExpr::try_from(context)` right? But will this implication hold true in the future? Is it possible for the group by expression to change in the future to allow for queries w/ aliases?

It would be nice if, in the event of such a change, something about this code would let us know we need to update it too, like a compilation failure or test failure. Maybe this should have a `assert!(result_aliased_exprs.iter().all())` to check for this guarantee. Not necessary if we don't expect this assumption to change though, I suppose

Reply:
We already allow queries with aliases. However in the future we will have possible filters and projections after group bys. So yes I will add the assertion.